src/main/java/org/radarbase/appserver/service/QuestionnaireScheduleService.java

```diff
@@ -148,6 +143,10 @@ public Schedule generateScheduleUsingProjectIdAndSubjectId(String projectId, Str
     @Transactional
     public Schedule generateScheduleForUser(User user) {
         Protocol protocol = protocolGenerator.getProtocolForSubject(user.getSubjectId());
+        if (protocol == null) {
+            subjectScheduleMap.put(user.getSubjectId(), new Schedule());
+            return new Schedule();
```

Comment:
why not create once and store in a variable or just return `subjectScheduleMap.get(user.getSubjectId())` instead of instantiating a new Schedule